src/resolve/resolved-dns-stub.c

```diff
@@ -703,13 +703,13 @@ static void dns_stub_process_query(Manager *m, DnsStubListenerExtra *l, DnsStrea
 
         if (dns_type_is_obsolete(p->question->keys[0]->type)) {
                 log_debug("Got message with obsolete key type, refusing.");
-                dns_stub_send_failure(m, l, s, p, DNS_RCODE_NOTIMP, false);
+                dns_stub_send_failure(m, l, s, p, DNS_RCODE_FORMERR, false);
```

Comment:
Why don't you send REFUSED, when logged message is refusing?

Reply:
Paranoia is a good reason to me.

Reply:
Are they dangerous in any sense?